zebra-chain/src/test_utils/vectors.rs

```diff
@@ -0,0 +1,179 @@
+#![forbid(unsafe_code)]
+//! Network methods for fetching blockchain vectors.
+//!
+
+#[cfg(any(test, feature = "proptest-impl"))]
+use std::collections::BTreeMap;
+
+#[cfg(any(test, feature = "proptest-impl"))]
+use crate::{
+    block::Block,
+    parameters::Network,
+    serialization::{SerializationError, ZcashDeserializeInto},
+};
+
+#[cfg(any(test, feature = "proptest-impl"))]
+use zebra_test::vectors::{
+    BLOCK_MAINNET_1046400_BYTES, BLOCK_MAINNET_653599_BYTES, BLOCK_MAINNET_982681_BYTES,
+    BLOCK_TESTNET_1116000_BYTES, BLOCK_TESTNET_583999_BYTES, BLOCK_TESTNET_925483_BYTES,
+    CONTINUOUS_MAINNET_BLOCKS, CONTINUOUS_TESTNET_BLOCKS, MAINNET_BLOCKS,
+    MAINNET_FINAL_SAPLING_ROOTS, MAINNET_FINAL_SPROUT_ROOTS,
+    SAPLING_FINAL_ROOT_MAINNET_1046400_BYTES, SAPLING_FINAL_ROOT_TESTNET_1116000_BYTES,
+    TESTNET_BLOCKS, TESTNET_FINAL_SAPLING_ROOTS, TESTNET_FINAL_SPROUT_ROOTS,
+};
+
+/// Network methods for fetching blockchain vectors.
+#[cfg(any(test, feature = "proptest-impl"))]
+impl Network {
+    /// Returns true if network is of type Mainnet.
+    pub fn is_mainnet(&self) -> bool {
+        matches!(self, Network::Mainnet)
+    }
+
+    /// Returns iterator over blocks.
+    pub fn get_block_iter(&self) -> std::collections::btree_map::Iter<'static, u32, &'static [u8]> {
+        if self.is_mainnet() {
+            MAINNET_BLOCKS.iter()
+        } else {
+            TESTNET_BLOCKS.iter()
+        }
+    }
+
+    ///
+    pub fn get_block_map(&self) -> &BTreeMap<u32, &'static [u8]> {
+        if self.is_mainnet() {
+            &*zebra_test::vectors::MAINNET_BLOCKS
+        } else {
+            &*zebra_test::vectors::TESTNET_BLOCKS
+        }
+    }
+
+    /// Returns genesis block for chain.
+    pub fn get_gen_block(&self) -> std::option::Option<&&[u8]> {
+        if self.is_mainnet() {
+            MAINNET_BLOCKS.get(&0)
+        } else {
+            TESTNET_BLOCKS.get(&0)
+        }
+    }
+
+    /// Returns block bytes
+    pub fn get_block_bytes(
+        &self,
+        main_bytes: u32,
+        test_bytes: u32,
+    ) -> Result<Block, SerializationError> {
+        if self.is_mainnet() {
+            match main_bytes {
+                653_599 => BLOCK_MAINNET_653599_BYTES.zcash_deserialize_into(),
+                982_681 => BLOCK_MAINNET_982681_BYTES.zcash_deserialize_into(),
+                _ => Err(SerializationError::UnsupportedVersion(main_bytes)),
+            }
+        } else {
+            match test_bytes {
+                583_999 => BLOCK_TESTNET_583999_BYTES.zcash_deserialize_into(),
+                925_483 => BLOCK_TESTNET_925483_BYTES.zcash_deserialize_into(),
+                _ => Err(SerializationError::UnsupportedVersion(test_bytes)),
+            }
+        }
+    }
```

Comment:
```suggestion
    pub fn get_test_block(&self, main_height: u32, test_height: u32) -> Option<Block> {
        let block_bytes = match (self.is_mainnet(), main_height, test_height) {
            (true, 653_599, _) => BLOCK_MAINNET_653599_BYTES,
            (true, 982_681, _) => BLOCK_MAINNET_982681_BYTES,
            (false, 583_999, _) => BLOCK_TESTNET_583999_BYTES,
            (false, 925_483, _) => BLOCK_TESTNET_925483_BYTES,
            _ => return None
        };

        block_bytes.zcash_deserialize_into().ok()
    }
```

Reply:
now returns option instead of result, renamed to get_test_block. Same with get_block_sapling_roots_bytes, renamed to get_test_block_sapling_roots.